db/migrate/20150525081931_remove_duplicate_tokens.rb

```diff
@@ -0,0 +1,28 @@
+class RemoveDuplicateTokens < ActiveRecord::Migration
+  def up
+    # only managed hosts should have tokens
+    Token.where(:id => Token.joins("left outer join hosts on hosts.id = tokens.host_id").
+                             where("hosts.type != 'Host::Managed'").pluck('tokens.id')).delete_all
+
+    # deleting duplicate tokens
+    hosts_with_duplicate_tokens = Token.having('count(*) > 1').group(:host_id).select('host_id')
+    existing_tokens = hosts_with_duplicate_tokens.map(&:token).compact
```

Comment:
@domcleal using `offset(1)` means I assume the first token returned in the correct one. I don't mind doing this if we don't care which token is kept as long as there is only one but maybe I should change to `Host.where(:id => hosts_with_duplicate).map(&:token)`?